app/models/local_photo.rb

```diff
@@ -84,7 +88,76 @@ class LocalPhoto < Photo
     "SAMSUNG CAMERA PICTURES",
     "MINOLTA DIGITAL CAMERA"
   ]
-  
+
+  def self.s3_host_alias( public = false )
+    unless ( CONFIG.s3_public_host || CONFIG.s3_public_bucket )
+      return ( CONFIG.s3_host || CONFIG.s3_bucket )
+    end
+    public ?
+      ( CONFIG.s3_public_host || CONFIG.s3_public_bucket ) :
+      ( CONFIG.s3_host || CONFIG.s3_bucket )
+  end
+
+  def self.s3_region( public = false )
+    return CONFIG.s3_region unless CONFIG.s3_public_region
+    public ? CONFIG.s3_public_region : CONFIG.s3_region
+  end
+
+  def self.s3_bucket( public = false )
+    return CONFIG.s3_bucket unless CONFIG.s3_public_bucket
+    public ? CONFIG.s3_public_bucket : CONFIG.s3_bucket
+  end
+
+  def self.s3_permissions( public = false )
+    return "public-read" unless CONFIG.s3_public_bucket && CONFIG.s3_public_acl
+    public ? CONFIG.s3_public_acl : "public-read"
+  end
+
+  def s3_host_alias
+    LocalPhoto.s3_host_alias( self.s3_account )
+  end
+
+  def s3_region
+    LocalPhoto.s3_region( self.s3_account )
+  end
+
+  def s3_bucket
+    LocalPhoto.s3_bucket( self.s3_account )
+  end
+
+  def s3_permissions
+    LocalPhoto.s3_permissions( self.s3_account )
+  end
+
+  def set_s3_account
+    self.s3_account = self.is_public? ? "public" : nil
+  end
+
+  def is_public?
+    self.large_url ? !! self.large_url.match( LocalPhoto.s3_bucket( true ) ) :
```

Comment:
Why `large_url` instead of `original_url`? I guess it doesn't matter, but `original_url` is theoretically the one that absolutely must exist.

Reply:
I just tend to avoid original in most bases, but I agree that is the canonical version. I've updated it